src/examples/source/SmartPeakGUI.cpp

```diff
@@ -1186,20 +1190,37 @@ int main(int argc, char** argv)
           session_handler_.setFeatureMatrix(application_handler_.sequenceHandler_);
           Eigen::Tensor<float, 2> x_data = session_handler_.feat_sample_data.shuffle(Eigen::array<Eigen::Index, 2>({ 1,0 }));
           Eigen::Tensor<float, 2> y_data = session_handler_.feat_value_data.shuffle(Eigen::array<Eigen::Index, 2>({ 1,0 }));
-          LinePlot2DWidget plot2d(x_data, y_data, session_handler_.feat_heatmap_row_labels, session_handler_.feat_line_x_axis_title, session_handler_.feat_line_y_axis_title,
+          LinePlot2DWidget plot2d(x_data, y_data, session_handler_.feat_col_labels, session_handler_.feat_row_labels, session_handler_.feat_line_x_axis_title, session_handler_.feat_line_y_axis_title,
             session_handler_.feat_line_sample_min, session_handler_.feat_line_sample_max, session_handler_.feat_value_min, session_handler_.feat_value_max,
             win_size_and_pos.bottom_and_top_window_x_size_, win_size_and_pos.top_window_y_size_, "FeaturesLineMainWindow");
           plot2d.draw();
           ImGui::EndTabItem();
         }
         if (show_feature_heatmap_plot && ImGui::BeginTabItem("Features (heatmap)", &show_feature_heatmap_plot))
         {
-          session_handler_.setMinimalDataAndFilters(application_handler_.sequenceHandler_);
-          session_handler_.setFeatureMatrix(application_handler_.sequenceHandler_);
-          Heatmap2DWidget plot2d(session_handler_.feat_heatmap_data, session_handler_.feat_heatmap_col_labels, session_handler_.feat_heatmap_row_labels,
-            session_handler_.feat_heatmap_x_axis_title, session_handler_.feat_heatmap_y_axis_title, session_handler_.feat_value_min, session_handler_.feat_value_max,
-            win_size_and_pos.bottom_and_top_window_x_size_, win_size_and_pos.top_window_y_size_, "FeaturesHeatmapMainWindow");
-          plot2d.draw();
+          if (!heatmap_plot_widget)
+          {
+            heatmap_plot_widget = std::make_unique<Heatmap2DWidget>(
+              session_handler_,
+              application_handler_.sequenceHandler_,
+              "Heatmap Main Window");
+          }
+          // To be replace when we have implement events
+          static bool heatmap_initialized = false;
+          if (!workflow_is_done_)
```

Comment:
If I understand this correctly, the user would not be able to view the heatmap while the workflow is running?  Is this restriction needed?

Reply:
The heatmap can be displayed, user can open and display the heatmap during the run. but if it is displayed during a running workflow, the data will be refreshed only when the workflow will be finished. said another way, it's just a means to notify the heatmap that the data has changed, do it can be updated automatically.